### ISSUE TITLE
Add support for multiple hook files per hook

### ISSUE THREAD

**Issue by strarsis** [roots/trellis#814]
## Submit a feature request or bug report

- [x] I've read the [guidelines for Contributing to Roots Projects](https://github.com/roots/guidelines/blob/master/CONTRIBUTING.md)
- [x] This is a feature request
- [ ] This is a bug report
- [x] This request isn't a duplicate of an [existing issue](https://github.com/roots/trellis/issues)
- [x] I've read the [docs](https://roots.io/trellis/docs) and followed them (if applicable)
- [x] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/c/trellis) forums

---

**What is the current behavior?**

Currently only one hook file can be declared per hook, 
notably also for hooks that already got a built-in Trellis hook file assigned 
which makes it harder to add and maintain additional custom hooks.

**What is the expected or desired behavior?**

Ability to specify more than one hook file, ideally as a list 
(with the built-in Trellis hooks already filled in by default).

**Is there a related [Discourse](https://discourse.roots.io/) thread or were any utilized (please link them)?**

https://discourse.roots.io/t/best-practices-extend-an-occupied-hook/9253

**Comment by swalkinshaw** [comment]
I think Phil's [reply](https://discourse.roots.io/t/best-practices-extend-an-occupied-hook/9253/3?u=swalkinshaw) solves this?

```diff
# deploy.yml
- deploy_build_after: "{{ playbook_dir }}/roles/deploy/hooks/build-after.yml"
+ deploy_build_after: "{{ playbook_dir }}/deploy-hooks/build-after.yml"
```

```yaml
# deploy-hooks/build-after.yml

- name: some task
  command: stuff

# original tasks
- include: "{{ playbook_dir }}/roles/deploy/hooks/build-after.yml"

- name: some task
  command: more stuff
```

**Comment by strarsis** [comment]
The YAML doesn't work and Trellis config has to be further modified, see https://discourse.roots.io/t/best-practices-extend-an-occupied-hook/9253/6?u=strarsis.

**Comment by swalkinshaw** [comment]
I'll re-open so we can make that adjustment

**Comment by strarsis** [comment]
A new approach that fixes the issue, is more elegant and allows to specify multiple hooks as list:
````
- include: "{{ item }}"
  with_items: "{{ deploy_build_after }}"
  when: deploy_build_after is defined
  tags: deploy-build-after
````

**Comment by fullyint** [comment]
@starsis thanks for your work on this. Building on your approach, this seems to work:
```
- include: "{{ item }}"
  with_items: "{{ deploy_build_after | default([]) }}"
  tags: deploy-build-after
```
`deploy_build_after` can be defined as a list but can also left alone as a simple string. So, fortunately users would _not_ have to change their hook definitions into lists (unless they want to).

If you're interested in submitting a PR adopting this implementation for deploy hooks, I'd support it.

**Comment by strarsis** [comment]
PR: https://github.com/roots/trellis/pull/815
First tests look good.

**Comment by fullyint** [comment]
Fixed/merged in #815 
Thanks @strarsis
